.builder/actions/mock_server_setup.py

```diff
@@ -0,0 +1,43 @@
+"""
+Setup local mock server for tests
+"""
+
+import Builder
+
+import os
+import sys
+import subprocess
+import atexit
+
+
+class MockServerSetup(Builder.Action):
+    """
+    Set up this machine for running the mock server test
+
+    This action should be run in the 'pre_build_steps' or 'build_steps' stage.
+    """
+
+    def run(self, env):
+        self.env = env
+        python_path = env.config['variables'].get('python')
+        # install dependency for mock server
+        result = self.env.shell.exec(python_path,
+                                     '-m', 'pip', 'install', 'h11', 'trio')
+        if result.returncode != 0:
+            print(
+                "Mock server failed to setup, skip the mock server tests.", file=sys.stderr)
+            exit(-1)
```

Comment:
you can do `self.env.shell.exec(..., check=True)`

and then you don't need to add any extra code for checking the return code